src/core/Entry.cpp

```diff
@@ -755,10 +755,10 @@ void Entry::addHistoryItem(Entry* entry)
     emitModified();
 }
 
-void Entry::removeHistoryItems(const QList<Entry*>& historyEntries)
+int Entry::removeHistoryItems(const QList<Entry*>& historyEntries)
```

Comment:
This function is all sorts of strange. It accepts a random list of entries and attempts to remove them from the internal history list and... even if it doesn't own that entry in history.... it deletes it (note this is only caught when asserts are enabled). This needs to be totally rethought from an api perspective.